experimental/dds/tree2/src/feature-libraries/sequence-field/moveEffectTable.ts

```diff
@@ -230,219 +232,222 @@ export function applyMoveEffectsToMark<T>(
 	consumeEffect: boolean,
 	composeChildren?: (a: T | undefined, b: TaggedChange<T>) => T | undefined,
 ): Mark<T>[] {
-	if (isAttachAndDetachEffect(mark)) {
-		if (isMoveIn(mark.attach)) {
-			if (isMoveOut(mark.detach)) {
-				// Move effects should not be applied to intermediate move locations.
-				return [mark];
-			}
-			const attachRevision = mark.attach.revision ?? revision;
+	return applyMoveEffectsToActiveMarks<T>(
+		applyMoveEffectsToVestigialMarks<T>(
+			[mark],
+			effects,
+			revision,
+			consumeEffect,
+			composeChildren,
+		),
+		revision,
+		effects,
+		consumeEffect,
+		composeChildren,
+	);
+}
+
+function applyMoveEffectsToVestigialMarks<T>(
+	inputQueue: Mark<T>[],
+	effects: MoveEffectTable<T>,
+	revision: RevisionTag | undefined,
+	consumeEffect: boolean,
+	composeChildren: ((a: T | undefined, b: TaggedChange<T>) => T | undefined) | undefined,
+): Mark<T>[] {
+	const outputQueue: Mark<T>[] = [];
+	let mark = inputQueue.shift();
+	while (mark !== undefined) {
+		const vestige = tryGetVestigialEndpoint(mark);
+		if (vestige !== undefined) {
 			const effect = getMoveEffect(
 				effects,
-				CrossFieldTarget.Destination,
-				attachRevision,
-				mark.attach.id,
+				CrossFieldTarget.Source,
+				vestige.revision ?? revision,
+				vestige.localId,
 				mark.count,
 			);
-
 			if (effect.length < mark.count) {
 				const [firstMark, secondMark] = splitMark(mark, effect.length);
-				const updatedAttach = firstMark.attach as MoveIn;
-				applyMoveEffectsToDest(
-					updatedAttach,
-					firstMark.count,
-					attachRevision,
+				mark = firstMark;
+				inputQueue.push(secondMark);
```

Comment:
unshift

Reply:
Turns out it doesn't matter because the queue is always empty when we add the second mark. I made the change (because it's more conceptually correct) and added test coverage for multi-splitting anyway.

Reply:
That's true for `applyMoveEffectsToActiveMarks` because it's called first with a queue of length 1 and the input queue length monotonically decreases over the course of the loop, but I think it's not necessarily true in general, right?

Reply:
Ah, that's right but flipped: `applyMoveEffectsToVestigialMarks` is the one that's always called with a queue of 1.